### PR TITLE
Add better description of what to pass to `hostfxr_initialize_for_dotnet_command_line`

### PR DESCRIPTION
**Pull request by vitek-karas**

### CODE REVIEW

File: docs/design/features/native-hosting.md [hunk 1]
```diff
@@ -235,7 +235,7 @@ int hostfxr_initialize_for_dotnet_command_line(
 
 Initializes the hosting components for running a managed application.
 The command line is parsed to determine the app path. The app path will be used to locate the `.runtimeconfig.json` and the `.deps.json` which will be used to load the application and its dependent frameworks.
-* `argc` and `argv` - the command line for running a managed application. These represent the arguments which would have been passed to the muxer if the app was being run from the command line.
+* `argc` and `argv` - the command line for running a managed application. These represent the arguments which would have been passed to the muxer if the app was being run from the command line. These are the parameters which are only valid for the runtime installation - SDK/CLI commands are not supported. So for example `app.dll app_argument_1 app_argument_2`. This specifically doesn't support the `dotnet run` SDK command.
```
Comment (elinor-fung) [inline docs/design/features/native-hosting.md:238]:
```suggestion
* `argc` and `argv` - the command line for running a managed application. These represent the arguments which would have been passed to the muxer if the app was being run from the command line. These are the parameters which are valid for the runtime installation by itself - SDK/CLI commands are not supported. For example, the arguments could be `app.dll app_argument_1 app_argument_2`. This API specifically doesn't support the `dotnet run` SDK command.
```

File: docs/design/features/native-hosting.md [hunk 2]
```diff
@@ -537,7 +537,7 @@ params.dotnet_root = get_directory(get_directory(get_directory(hostfxr_path)));
 hostfxr_handle host_context_handle;
 hostfxr_initialize_for_dotnet_command_line(
     _argc_,
-    _argv_,
+    _argv_,  // Should be for example 'app.dll app_argument_1 app_argument_2'
```
Comment (elinor-fung) [inline docs/design/features/native-hosting.md:540]:
```suggestion
    _argv_,  // For example, 'app.dll app_argument_1 app_argument_2'
```

**Review by elinor-fung — APPROVED** [review]
Do we want to update the header doc too? It mentions not supporting SDK commands, but the example might be helpful there too, since we do ship that header in our packs.
https://github.com/dotnet/runtime/blob/0fcd7690a14335a8807144d0e7ad937aa83699b0/src/native/corehost/hostfxr.h#L104

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by vitek-karas** [comment]
This change only modifies a comment, so the test failure is definitely not related. Filed https://github.com/dotnet/runtime/issues/53720 to track it.

**Pull request merged**